src/ObjectGraph.es6.js

```diff
@@ -54,6 +54,14 @@ ObjectGraph.prototype.init = function(opts) {
 
   // Try to prevent recursion into internal structures.
   this.blacklistedObjects.push(this);
+
+  // Lock-in user agent by (potentially) copying it into an own property.
```

Comment:
Why is just potentially? This unconditionally assigns `this.userAgent`, but is there some way it could have already been an own property? Is there still a reason to have `ObjectGraph.prototype.userAgent` now, or could have just `this.userAgent`, like for `environment`? That'd be easier to understand I think.

Reply:
The reason I would leave the prototype value in place is defensive: If we deserialize an `ObjectGraph` with no `userAgent`, it should assume its default value.

Reply:
As discussed on VC, since ObjectGraph instances are used both when collecting data on the browser and are also created with `ObjectGraph.fromJSON` in node when processing the data, there's a risk of confusion if the node UA string shows up there from the prototype. So setting `userAgent` only in a code path that's run when collecting data would be best. (Can't quite tell if this is such a code path.)